### DISCUSSION TITLE
How to make a note have a different title to its filename?

### DISCUSSION THREAD

**Discussion by ckp95** [srid/neuron#550] [General]
Trying out this program. I want to be able to link to a note without having to use its filename. This is because I may want long titles, titles with spaces, titles with weird Unicode, or other things that might make life difficult down the road if they were used as filenames. I just want to have the filenames be meaningless and have the actual metadata inside the file. Fulltext search > filenames.

So I go through the tutorial and it says to make a file with `neuron new -e`, so I do that and make this file with the randomly generated filename `cb3bc096.md`:

```
---
date: 2021-02-01T13:42
title: A test note
---

# Heading of the note

Body of the note
```

Then I make a `index.md` file with these contents:

```
# My Zettelkasten 

From here, you can visit: [[A test note]]
```

And I generate the site with `neuron gen -wS` like the docs say, except that doesn't work and I need to do `neuron rib -wS` instead for some reason. Anyway I do that and it doesn't work. It just says "links to non-existent zettel: A test note".

So I think maybe I need to use the "slug" field. I add `slug: A test note` to cb3bc096.md. Doesn't work either. I try with hyphens instead of spaces, doesn't work.

So .. how do I make these kinds of links? Surely I'm not meant to memorize these gibberish autogenerated filenames? I don't get it.

**Comment by ckp95** [comment]
Also is it me or is the documentation site really, really slow to load for just being static html?

**Comment by ckp95** [comment]
Oh wait I get it now, I can just give a short title like `test-note.md` and then a longer title with whatever formatting I need by doing `title: A test note`. Then linking to it is just `[[test-note]]` and it renders with the title. That makes sense, although the docs could be clearer.

Although now I tried changing the slug part to be `slug: quick-brown-fox` yet the HTML filename is still `test-note.html`, so I don't understand that part.

And the locally hosted site is really really slow just like the web hosted docs. I don't understand why this is. A page with two lines of content takes 2-4 seconds to load fully.

**Comment by srid** [comment]
The docs can certainly be improved; PR welcome on that.

> Although now I tried changing the slug part to be slug: quick-brown-fox yet the HTML filename is still test-note.html, so I don't understand that part.

Note that `neuron gen` doesn't automatically delete stale .html files. So the old files linger. Just visit `quick-brown-fox.html`, or go back to Impulse and visit the note from there. see https://github.com/srid/neuron/issues/141

> And the locally hosted site is really really slow

No one else has reported this.

**Comment by srid** [comment]
About your original question, just do this:

```
$ echo "Hello world" > "A test note.md"
```

And link to it as `[[A test note]]`. No need for `title` YAML meta. 

This is what is known as "title IDs". See https://neuron.zettel.page/id.html
